src/main/java/com/buyanywhere/productcatalog/controllers/v1/CategoryController.java

```diff
@@ -1 +1,40 @@
+package com.buyanywhere.productcatalog.controllers.v1;
 
+import com.buyanywhere.productcatalog.exceptions.CategoryNotFoundException;
+import org.springframework.web.bind.annotation.*;
+import com.buyanywhere.productcatalog.models.Category;
+import com.buyanywhere.productcatalog.repositories.CategoryRepository;
+
+@RestController
+@RequestMapping("/v1/category")
+
+public class CategoryController {
+
+    private CategoryRepository repository;
+
+    public CategoryController(CategoryRepository repository) {
+        this.repository = repository;
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "/")
+    public Category post(@RequestBody Category category) {
+        return repository.save(category);
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/{id}")
+    public Category get(@PathVariable("id") long id) throws CategoryNotFoundException {
+        if (!repository.findById(id).isPresent() || repository.findById(id).get().isDeleted()) {
+            throw new CategoryNotFoundException(id);
+        }
+        return repository.findById(id).get();
+    }
+
+    @RequestMapping(method = RequestMethod.PATCH, value = "/{id}/{name}/{displayOrder}")
```

Comment:
The Patch verb in this case is correct, but imagine a class that has 20 fields, if you use PATCH and send all the fields in the path variables the resulting URL will be really big.
 I would rather have you choose PUT, since with this verb you can send a JSON in the request's body and receive a Category in this method and save it using the repository remember to validate if the category already exists, if it doesn't then you should throw an exception. Also you must validate that name and displayOrder have correct data. 
The name of the method should not be update (the name is not incorrect) it should be put.